File: C++ for Everyone, 2nd Ed, Horstmann/Chapter4/P4-3E.cpp
//Write programs that read a line of input as a string and print
//e.The positions of all vowels in the string.

#include<iostream>
#include<vector>

using namespace std;

int main()
{
    cout << "Enter a string: ";
    string input;
    getline(cin, input);

    vector<char> inputCstyle (input.length() + 1);
    copy(input.begin(), input.end(), inputCstyle.begin());

    int nVowels = 0;
    for(int i = 0; i < input.length(); i++)
    {
        if(inputCstyle[i] == 'A' || inputCstyle[i] == 'a' || inputCstyle[i] == 'E' || inputCstyle[i] == 'e' || inputCstyle[i] == 'O' || inputCstyle[i] == 'o' || inputCstyle[i] == 'U' || inputCstyle[i] == 'u' || inputCstyle[i] == 'I' || inputCstyle[i] == 'i')
        {
            cout << "Vowel at [" << i << "]" << "\n";
        }
    }
    return 0;
}

File: Python For Everyone, Horstmann/Chapter4-Loops/P4.3E.py
# Write programs that read a line of input as a string and print
#     a.  Only the uppercase letters in the string.
#     b.  Every second letter of the string.
#     c.  The string, with all vowels replaced by an underscore.
#     d.  The number of digits in the string.
#     e.  The positions of all vowels in the string.

inputString = str(input("Enter a string: "))
positions = ""
# saving time writing long statements
vowels = ('a','e','i','o','u','A','E','I','O','U')

for i in range(len(inputString)):
    if inputString[i] in vowels:
            positions += str(i) + "-"


# fix for the last dash in the positions string
positions = positions[:-1]
print(positions)